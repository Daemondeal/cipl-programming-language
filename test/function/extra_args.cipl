proc f(a, b):
    println(a)
    println(b)

f(1, 2, 3, 4)