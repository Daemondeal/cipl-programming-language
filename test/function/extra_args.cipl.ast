(proc f (params a b) (body
    (expression (call print a))
    (expression (call print b))
)
(expression (call f 1 2 3 4))

