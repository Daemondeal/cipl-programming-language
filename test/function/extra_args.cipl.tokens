Token          Lexeme         Line 
PROC           proc           1    
IDENTIFIER     f              1    
LEFT_PAREN     (              1    
IDENTIFIER     a              1    
COMMA          ,              1    
IDENTIFIER     b              1    
RIGHT_PAREN    )              1    
COLON          :              1    
LINE_END       \n             2    
INDENT                        2    
IDENTIFIER     println        2    
LEFT_PAREN     (              2    
IDENTIFIER     a              2    
RIGHT_PAREN    )              2    
LINE_END       \n             3    
IDENTIFIER     println        3    
LEFT_PAREN     (              3    
IDENTIFIER     b              3    
RIGHT_PAREN    )              3    
LINE_END       \n             4    
DEDENT                        5    
IDENTIFIER     f              5    
LEFT_PAREN     (              5    
NUMBER         1              5     1
COMMA          ,              5    
NUMBER         2              5     2
COMMA          ,              5    
NUMBER         3              5     3
COMMA          ,              5    
NUMBER         4              5     4
RIGHT_PAREN    )              5    
LINE_END       \n             5    
EOF                           5    
