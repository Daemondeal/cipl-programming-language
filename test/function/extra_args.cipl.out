::stdout::

::stderr::
Error: Expected 2 arguments but got 4 for <proc f>() at line 5
